src/SessionSetup/RetrieveInstancesWidgetDemo/main.cpp

```diff
@@ -3,22 +3,55 @@
 // found in the LICENSE file.
 
 #include <QApplication>
+#include <QMessageBox>
 #include <memory>
 
 #include "OrbitBase/Result.h"
 #include "OrbitGgp/Client.h"
+#include "QtUtils/MainThreadExecutorImpl.h"
+#include "SessionSetup/RetrieveInstances.h"
 #include "SessionSetup/RetrieveInstancesWidget.h"
 
+namespace {
+constexpr const char* kOrganizationName = "The Orbit Authors";
+constexpr const char* kApplicationName = "RetrieveInstancesWidgetDemo";
+}  // namespace
+
 using orbit_ggp::Client;
+using orbit_session_setup::RetrieveInstances;
+using orbit_session_setup::RetrieveInstancesWidget;
 
 int main(int argc, char* argv[]) {
   QApplication app{argc, argv};
+  QCoreApplication::setOrganizationName(kOrganizationName);
+  QCoreApplication::setApplicationName(kApplicationName);
 
   ErrorMessageOr<std::unique_ptr<Client>> client_or_error = orbit_ggp::CreateClient();
   FAIL_IF(client_or_error.has_error(), "%s", client_or_error.error().message());
+  Client* client_ptr = client_or_error.value().get();
+
+  std::shared_ptr<orbit_qt_utils::MainThreadExecutorImpl> executor{
+      orbit_qt_utils::MainThreadExecutorImpl::Create()};
+
+  std::unique_ptr<RetrieveInstances> retrieve_instances =
+      RetrieveInstances::Create(client_ptr, executor.get());
+
+  RetrieveInstancesWidget widget{executor.get(), retrieve_instances.get()};
+
+  QObject::connect(&widget, &RetrieveInstancesWidget::LoadingSuccessful, &widget,
+                   [&widget](const QVector<orbit_ggp::Instance>& instances) {
+                     auto message =
```

Comment:
Nit: In this case I'd specify the type instead of `auto` for readability as it's super short.

Reply:
It was actually a deliberate decision, because the type of the variable can be easily inferred from the same line. Similar how we use `auto ptr = std::make_unique<Type>()`. But you do have point of QString being very short and the only real benefit of auto is avoiding boilerplate. I changed it.